analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/PermissiveCors.cs

```diff
@@ -0,0 +1,145 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Extensions;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.Hotspots
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class PermissiveCors : TrackerHotspotDiagnosticAnalyzer<SyntaxKind>
+    {
+        private const string DiagnosticId = "S5122";
+        private const string MessageFormat = "Make sure this permissive CORS policy is safe here.";
+        private const string AccessControlAllowOriginHeader = "Access-Control-Allow-Origin";
+        private const string AccessControlAllowOriginPropertyName = "AccessControlAllowOrigin";
+        private const string StarConstant = "*";
+        private const int ParameterCount = 2;
+
+        protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+        public PermissiveCors() : base(AnalyzerConfiguration.Hotspot, DiagnosticId, MessageFormat) { }
+
+        public PermissiveCors(IAnalyzerConfiguration configuration) : base(configuration, DiagnosticId, MessageFormat) { }
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            base.Initialize(context);
+
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                if (IsEnabled(compilationContext.Options))
+                {
+                    context.RegisterSyntaxNodeActionInNonGenerated(VisitAttribute, SyntaxKind.Attribute);
+                }
+            });
+        }
+
+        protected override void Initialize(TrackerInput input)
+        {
+            var iTracker = Language.Tracker.Invocation;
```

Comment:
`iTracker` sounds like an interface... maybe we can find a better name?

`invocationTracker` may be longer, but it doesn't sound like an interface :)

or if you split the Invocation tracking from Object Creation tracking in two methods, you'll be able to use just tracker, or just `it`